File: 349. Intersection of Two Arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        rlist = []
        for i in range(len(nums1)):
            for j in range(len(nums2)):
                if(nums1[i] == nums2[j]):
                    if(rlist.count(nums1[i]) == False):
                        rlist.append(nums1[i])
        return rlist;

if __name__ == '__main__':
    nums1 = [1,2,2,1]
    nums2 = [2,2]
    result = Solution().intersection(nums1,nums2)
    print result

File: 349. Intersection of Two Arrays.java
package binarySearch;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class Solution {
	/**
	 * @param nums1 [1, 2, 2, 1]
	 * @param nums2 [2, 2]
	 * @return
	 */
	public int[] intersection(int[] nums1, int[] nums2) {
		Set<Integer> set = new HashSet<>();
		Arrays.sort(nums1);
		Arrays.sort(nums2);
		int i = 0;
		int j = 0;
		while(i < nums1.length && j < nums2.length) {
			if(nums1[i] < nums2[j])
				i++;
			else if(nums1[i] > nums2[j])
				j++;
			else {
				set.add(nums1[i]);
				i++;
				j++;
			}
		}
		int[] result = new int[set.size()];
		int k = 0;
		for(Integer num : set) {
			result[k++] = num;
		}
		return result;
		
	}

	public static void main(String[] args) {
		int[] nums1 = {1,2,2,1};
		int[] nums2 = {2,2};
		new Solution().intersection(nums1,nums2);
		int[] result = new Solution().intersection(nums1,nums2);
		for(int num: result) {
			System.out.println(num + " ");
		}
	}

}
